### PR TITLE
First implementation

### PR DESCRIPTION
**Pull request by hahleung**
Development of the scrabble exercise: choose the most valuable word from a random draw.

### CODE REVIEW

File: hello.rb [hunk 1]
```diff
@@ -0,0 +1 @@
+puts "it works!"
```
Comment (ostapneko) [inline hello.rb]: You should kill this file

File: scrabble.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	 if x < counter
+	   stack[x] = LETTERS[current_counter]
+	 else
+	   current_counter = current_counter + 1
+	   counter = counter + QUANTITY_OF_LETTERS[current_counter]
+	   stack[x] = LETTERS[current_counter]
+	 end
+	end
+	stack
+	#Only last sentence of a function is returned.
```
Comment (ostapneko) [inline scrabble.rb]: Remove the comments

File: scrabble.rb [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  new_combinations = combinations.map { |e| combine(e) }.flatten(1) + [draw_input]
+  new_combinations.uniq.reject(&:empty?)
+  #Also working: reject(x==[])
+  #Also working: reject(|x| x.empty?)
+end
+
+#PART 5: Import existing words database and purge the set of combinations
+#http://www.freescrabbledictionary.com/sowpods/download/sowpods.txt
+def words
+	words_raw = File.readlines("words.txt")
```
Comment (ostapneko) [inline scrabble.rb]: Use spaces not tabs

File: scrabble.rb [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	words_raw.map { |x| x.delete("\n") }
+end
+
+def existing_words
```
Comment (ostapneko) [inline scrabble.rb]: use `valid_words` instead?

File: scrabble.rb [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+	words_current = words
+	draw_current = draw
+
+	p "The 7 letters drawn are:"
+	p draw_current
+
+	combinations = combine(draw_current).map { |e| e.join }
+	combinations.map { |x| x if words_current.include?(x) }.compact
```
Comment (ostapneko) [inline scrabble.rb]: You should use filter instead

Comment (ostapneko) [inline scrabble.rb]: Could you find a much quicker algorithm?

File: hello.rb (hunk 1 above)
Comment (hahleung) [inline hello.rb]: Done

File: scrabble.rb (hunk 2 above)
Comment (hahleung) [inline scrabble.rb]: Done

File: scrabble.rb (hunk 4 above)
Comment (hahleung) [inline scrabble.rb]: Done: 'words_raws' -> 'valid_words'

**Pull request closed without merging**